src/gateway/router/routes/registerRoute.ts

```diff
@@ -0,0 +1,108 @@
+import Router from '@koa/router';
+import { evalType } from '../../tasks/contractsMetadata';
+import { BUNDLR_NODE2_URL } from '../../../constants';
+import { DataItem } from 'arbundles';
+import rawBody from 'raw-body';
+import { Tag } from 'arweave/node/lib/transaction';
+import { sleep } from 'warp-contracts';
+import { updateCache } from '../../updateCache';
+
+export async function registerRoute(ctx: Router.RouterContext) {
+  const { logger, gatewayDb, arweave, bundlr } = ctx;
+
+  const rawDataItem: Buffer = await rawBody(ctx.req);
+  const dataItem = new DataItem(rawDataItem);
+
+  try {
+    const isValid = await dataItem.isValid();
+    if (!isValid) {
+      throw new Error(`Data item binary is not valid.`);
+    }
+
+    const areContractTagsValid = await verifyContractTags(dataItem);
+    if (!areContractTagsValid) {
+      throw new Error(`Contract tags are not valid.`);
+    }
+
+    const bundlrResponse = await bundlr.uploader.transactionUploader(dataItem);
+
+    if (
+      bundlrResponse.status !== 200 ||
+      !bundlrResponse.data.public ||
+      !bundlrResponse.data.signature ||
+      !bundlrResponse.data.block
+    ) {
+      throw new Error(
+        `Bundlr did not upload transaction correctly. Bundlr responded with status ${bundlrResponse.status}.`
+      );
+    }
+
+    logger.debug('Data item successfully bundled.', {
+      id: bundlrResponse.data.id,
+    });
+
+    const srcTxId = dataItem.tags.find((d) => d.name == 'Contract-Src')!.value;
+    const initStateRaw = dataItem.tags.find((d) => d.name == 'Init-State')!.value;
+    const initState = JSON.parse(initStateRaw);
+    const type = evalType(initState);
+    const ownerAddress = await arweave.wallets.ownerToAddress(bundlrResponse.data.public);
```

Comment:
Noooo!!! `bundlrResponse.data.public` is the jwk.n of the Bundlr node, not the jwk.n of the signed dataitem (and I guess we want to set here the DataItem's signer jwk.n, no the Bundlr's one)